inventory/environ.py

```diff
@@ -182,19 +181,13 @@ def overrideEnvironmentVars(vars_scope):
         vars_scope["splunk"]["idxc"] = {}
     vars_scope["splunk"]["idxc"]["secret"] = os.environ.get('SPLUNK_IDXC_SECRET', vars_scope["splunk"]["idxc"]["secret"])
     vars_scope["splunk"]["enable_service"] = os.environ.get('SPLUNK_ENABLE_SERVICE', vars_scope["splunk"]["enable_service"])
+    vars_scope["splunk"]["allow_upgrade"] = os.environ.get('SPLUNK_ALLOW_UPGRADE', vars_scope["splunk"]["allow_upgrade"])
```

Comment:
Pedantic, perhaps, but I think this should default to `True`, perhaps being renamed to `SPLUNK_DISALLOW_UPGRADE`.

Reply:
I have personal preference for a positive flag instead of a negative one. But this set aside I agree with you that in an automation system, the upgrade should by default be automatic.
The flag should be kept in order to have the ability to block certain nodes from upgrading while others do.

if to call it allow or disallow... I leave that up to the majority to decide.